tests/store/api/find/test_find_business_data_case.py

```diff
@@ -0,0 +1,117 @@
+"""Tests the findbusinessdata part of the Cg store API related to the Family model."""
+from datetime import datetime
+from typing import List
+from cg.constants.constants import CaseActions
+from cg.store import Store
+from cg.store.models import (
+    Customer,
+    Family,
+)
+from tests.store_helpers import StoreHelpers
+from cg.constants import Pipeline
+
+
+def test_get_filtered_cases_no_filters(store_with_cases_and_customers: Store):
+    """Test that all cases are returned when no filters are applied."""
+    # GIVEN a store with some cases
+    all_cases_query = store_with_cases_and_customers._get_query(table=Family)
+    assert all_cases_query.count() > 0
+
+    # WHEN calling get_filtered_cases with no filters
+    filtered_cases_query = store_with_cases_and_customers.get_filtered_cases()
+
+    # THEN all cases should be returned
+    assert filtered_cases_query.count() == all_cases_query.count()
```

Comment:
So this behaviour seems that it should just be a function like.

```Python
def get_cases(self):
   return self._get_query(Table=Family).all()
```

Otherwise the purpose of applying filter functions is defeated. And in this case, there is no `filtered_cases` but simply all, which makes the function behave differently with each input.

Reply:
So your point is that the arguments to the get_filtered_cases should not be optional? Otherwise, it makes sense to return all cases when no filters are provided (since no filtering can be done).

Reply:
I think these are the sub functions that can be extracted. Filter by:

- Customer and case id search pattern
- Customers, action and case id/name pattern
- Customers, pipeline and case id/name pattern

But I think I prefer the current function.